certinject/cryptoapi_windows.go

```diff
@@ -0,0 +1,181 @@
+package certinject
+
+import "golang.org/x/sys/windows/registry"
+import "crypto/sha1"
+import "encoding/hex"
+import "strings"
+import "math"
+import "time"
+
+// In 64-bit Windows, this key is shared between 64-bit and 32-bit applications.
+// See https://msdn.microsoft.com/en-us/library/windows/desktop/aa384253.aspx
+const cryptoApiCertStoreRegistryBase = registry.LOCAL_MACHINE
+const cryptoApiCertStoreRegistryKey = `SOFTWARE\Microsoft\EnterpriseCertificates\Root\Certificates`
+const cryptoApiMagicName = "Namecoin"
+const cryptoApiMagicValue = 1
+
+func injectCertCryptoApi(derBytes []byte) {
+
+	// Format documentation of Microsoft's "Certificate Registry Blob":
+	
+	// 5c 00 00 00 // propid
+	// 01 00 00 00 // unknown
+	// 04 00 00 00 // size (little endian because Microsoft engineers suck at their jobs)
+	// subject public key bit length // data[size]
+	
+	// 19 00 00 00
+	// 01 00 00 00
+	// 10 00 00 00
+	// MD5 of ECC pubkey of certificate
+	
+	// 0f 00 00 00
+	// 01 00 00 00
+	// 20 00 00 00
+	// Signature Hash
+	
+	// 03 00 00 00
+	// 01 00 00 00
+	// 14 00 00 00
+	// Cert SHA1 hash
+	
+	// 14 00 00 00
+	// 01 00 00 00
+	// 14 00 00 00
+	// Key Identifier
+	
+	// 04 00 00 00
+	// 01 00 00 00
+	// 10 00 00 00
+	// Cert MD5 hash
+	
+	// 20 00 00 00
+	// 01 00 00 00
+	// cert length
+	// cert
+	
+	// But, guess what?  All you need is the "20" record.
+	// Windows will happily regenerate all the others for you, whenever you actually try to use the certificate.
+	// How cool is that?
+	
+	// Length of cert
+	certLength := len(derBytes)
+	
+	// Header for a stripped Windows Certificate Registry Blob
+	certBlobHeader := []byte{0x20, 0, 0, 0, 0x01, 0, 0, 0, byte( (certLength >> 0) & 0xFF), byte( (certLength >> 8) & 0xFF), byte( (certLength >> 16) & 0xFF), byte( (certLength >> 24) & 0xFF) }
+	
+	// Construct the Blob
+	certBlob := append(certBlobHeader, derBytes...)
+	
+	// Open up the cert store.
+	certStoreKey, err := registry.OpenKey(cryptoApiCertStoreRegistryBase, cryptoApiCertStoreRegistryKey, registry.ALL_ACCESS)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Actually, I'm trying to remember why I used Fatal instead of Error and I suspect I made a mistake.  Do you agree that Errore makes more sense than Fatale?